nessie/src/main/java/org/apache/iceberg/nessie/NessieUtil.java

```diff
@@ -165,4 +180,77 @@ public static TableMetadata updateTableMetadataWithNessieSpecificProperties(
 
     return builder.discardChanges().build();
   }
+
+  static void handleExceptionsForCommits(
+      Exception exception, String refName, AtomicBoolean failure, Content.Type type) {
+    if (exception instanceof NessieConflictException) {
+      failure.set(true);
```

Comment:
This looks awkward to me. Could this code be refactored somehow to avoid out parameters? Maybe add a status class with multiple fields?

Reply:
removed `AtomicBoolean` as it can be simplified.